### PR TITLE
fix: docker superset home and datadir

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Fixes superset home directory on docker, associated issue #14867

Superset actually tries to create superset home:
https://github.com/apache/superset/blob/b38596fd961ae3317554dd79984df710601aa135/superset/app.py#L94

### ADDITIONAL INFORMATION
- [ ] Has associated issue: #14867
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -85,6 +85,7 @@ ENV LANG=C.UTF-8 \
     SUPERSET_PORT=8088
 
 RUN useradd --user-group -d ${SUPERSET_HOME} --no-log-init --shell /bin/bash superset \
+        && mkdir -p ${SUPERSET_HOME} \
```
Comment (dungdm93) [inline Dockerfile:88]: Why not use `useradd -m`. 
```
-m, --create-home             create the user's home directory
```

Comment (dpgaspar) [reply Dockerfile:88]: because /app only allows `root` and superset user is not created yet, so that fails

Comment (dpgaspar) [reply Dockerfile:88]: if fails with:
```
 => ERROR [lean  2/11] RUN useradd --user-group -m -d /app/superset_home --no-log-init --shell /bin/bas  0.3s
------
 > [lean  2/11] RUN useradd --user-group -m -d /app/superset_home --no-log-init --shell /bin/bash superset         && mkdir -p /app/pythonpath         && apt-get update -y         && apt-get install -y --no-install-recommends             build-essential             default-libmysqlclient-dev             libsasl2-modules-gssapi-mit             libpq-dev         && rm -rf /var/lib/apt/lists/*:
#6 0.304 useradd: cannot create directory /app/superset_home
```

**Review by willbarrett — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (shawnzhu) [reply Dockerfile:88]: :point_up: is caused by missing directory `/app`, which comes after this line but really should put before it:

https://github.com/apache/superset/blob/0eb34be78937edc509b6df872bc831d28add675f/Dockerfile#L88

see my PR: #14903

Comment (dungdm93) [reply Dockerfile:88]: Got it.
So I'd like to suggest that you need to change ownership of `${SUPERSET_HOME}` to `superset` as well.
```
chown -R superset: ${SUPERSET_HOME}`
```

Comment (ethack) [reply Dockerfile:88]: @shawnzhu's https://github.com/apache/superset/pull/14903 is in line with my testing for https://github.com/apache/superset/issues/14847#issuecomment-848961017 and avoids the issue with home directory ownership present in this PR.

Comment (dpgaspar) [reply Dockerfile:88]: @shawnzhu @ethack agree it's the missing PYTHONPATH. let's merge #14903, thank you

**Pull request closed without merging**